snapcraft/commands/daemon.start

```diff
@@ -625,12 +625,11 @@ for pid in $(pgrep -f "lxd(\.debug)? --logfile"); do
 
     # Confirm the PID found by pgrep is indeed ours and not one from
     # a nested instance also running LXD or a recycled PID
-    rm -f "/var/snap/lxd/common/lxd/.validate"
-    touch "/proc/${pid}/root/var/snap/lxd/common/lxd/.validate" 2>/dev/null || true
-    if [ -e "/var/snap/lxd/common/lxd/.validate" ]; then
+    echo "$$" > "/proc/${pid}/root/var/snap/lxd/common/lxd/.validate" 2>/dev/null || true
```

Comment:
I'm not really following why this change is needed (leaving aside the microoptimisation for now too).

What issue is this fixing? Have we seen it occur? 

I'm confused why a nested LXD would be able to write to the /var/snap/lxd/common/lxd directory of its parent?

Reply:
It's not fixing any known/observed issue more a theoretical/potential one.

This came about after I asked Aleks if it was OK to replace the touch+rm by just checking for `"/proc/${pid}/root/var/snap/lxd/common/lxd/server.key`'s presence.

Aleks quickly pointed out that would defeat the purpose of that check which is to make sure we are dealing with a LXD PID that belongs to our PID and mount NS. That's because any nested LXD PID we'd get from that `pgrep` would also lead us to a `server.key` but in a different mount NS. Hence the writing to it was needed.

I then observed that if the nested LXD was also going through that same validation at the same time, it could also race and create the `.validate` file to causing problems.

To which Aleks replied: "write and compare random data to eliminate the race".